components/supportivekoala/supportivekoala.app.mjs

```diff
@@ -1,11 +1,77 @@
+import { axios } from "@pipedream/platform";
+
 export default {
   type: "app",
   app: "supportivekoala",
-  propDefinitions: {},
+  propDefinitions: {
+    templateId: {
+      type: "string",
+      label: "Template ID",
+      description: "Template that defines how the image will be created",
+      async options() {
+        const templates = await this.getTemplates();
+        return templates.map((template) => ({
+          label: template.name,
+          value: template._id,
+        }));
+      },
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    async _makeRequest({
+      $ = this,
+      opts,
+    }) {
+      const {
+        method = "get",
+        path,
+        params,
+        data,
+      } = opts;
+      return axios($, {
+        headers: {
+          "Authorization": `Bearer ${this.$auth.api_key}`,
+          "Content-Type": "application/json",
+        },
+        url: `https://api.supportivekoala.com/v1${path}`,
+        method,
+        params,
+        data,
+      });
+    },
+    async getTemplates() {
+      return this._makeRequest({
+        opts: {
+          path: "/templates",
+        },
+      });    },
+    async getTemplate({
+      $ = this,
+      templateId,
+    }) {
+      return this._makeRequest({
+        $,
+        opts: {
+          path: `/templates/${templateId}`,
+        },
+      });
+    },
+    async createImage({
+      $,
+      templateId,
+      params,
+    }) {
+      return this._makeRequest({
+        $,
+        opts: {
+          path: "/images",
+          method: "post",
+          data: {
+            template: templateId,
+            params,
+          },
+        },
+      });
     },
```

Comment:
Same refactor here
```suggestion
    async createImage(args = {}) {
      return this._makeRequest({
        path: "/images",
        method: "post",
        ...args,
      });
    },
```
here it's not necessary to set `templateId` since you can pass it with `data` so when you call `createImage` you can do it like
```javascript
await this.createImage({
  data: {
    template,
    params,
  },
});
```